src/main/java/Duke.java

```diff
@@ -1,10 +1,264 @@
+            dateTime.setDateAndTime();
+        }
+
+        return dateTime;
+    }
+
+    public static void updateAdd(Task task) {
+        System.out.println("Got it. I've added this task:");
+        System.out.println("    " + task);
+        if (taskList.size() > 1) {
+            System.out.println("Now you have " + taskList.size() + " tasks in the list.");
+        } else {
+            System.out.println("Now you have " + taskList.size() + " task in the list.");
+        }
+    }
+
+    public static void updateRemove(Task task) {
+        System.out.println("Noted. I've removed this task:");
+        System.out.println("    " + task);
+
+        if (taskList.size() > 1) {
+            System.out.println("Now you have " + taskList.size() + " tasks in the list.");
+        } else {
+            System.out.println("Now you have " + taskList.size() + " task in the list.");
+        }
+    }
+
+
+    public static void list() {
+        if (taskList.size() > 0) {
+            for (int i = 0; i < taskList.size(); i++) {
+                int index = i + 1;
+                System.out.println(index + "." + taskList.get(i));
+            }
+        } else {
+            System.out.println("You have no tasks in the list.");
+        }
+    }
+
+    public static void done(String position) throws DukeException {
+        int index = Integer.parseInt(position) - 1;
+        if (index < taskList.size()) {
+            Task task = taskList.get(index);
+            task.setDone(true);
+        } else {
+            throw new DukeException("OOPS!!! The task you want to complete is not in the list!");
+        }
+    }
+
+    public static Task todo(String action) {
+        Task task = new Todo(action);
+        taskList.add(task);
+
+        return task;
+    }
+
+    public static Task deadline(String action, DateTime dateTime) {
+        Task task = new Deadline(action, dateTime);
+        taskList.add(task);
+
+        return task;
+    }
+
+    public static Task event(String action, DateTime dateTime) {
+        Task task = new Event(action, dateTime);
+        taskList.add(task);
+
+        return task;
+    }
+
+    public static Task delete(String position) throws DukeException {
+        int index = Integer.parseInt(position) - 1;
+        Task task;
+
+        if (index < taskList.size()) {
+            task = taskList.get(index);
+            taskList.remove(index);
+        } else {
+            throw new DukeException("OOPS!!! The task you want to remove is not in the list!");
+        }
+
+        return task;
+    }
+
+    public static void writeToFile(String filePath) throws IOException {
+        FileWriter fw = new FileWriter(filePath);
+        for (Task task : taskList) {
+            String text = task.getFileFormat();
+            fw.write(text + System.lineSeparator());
+        }
+        fw.close();
+    }
+
+    public static void readFileContents(String filePath) throws FileNotFoundException, DukeException {
```

Comment:
Maybe it would be better for you to catch the FileNotFoundException in this method and throw a DukeException instead? In this case you do not have to catch 2 types of exception in the method that calls this method.